src/dataServices/dynamoDbBundleService.ts

```diff
@@ -270,6 +286,81 @@ export class DynamoDbBundleService implements Bundle {
+                idToVersionId[`${itemRequest.resourceType}_${itemRequest.id}`] = `${itemRequest.vid + 1}`;
+            }
+        });
+
+        const createOrUpdates = requests.filter((request: BatchReadWriteRequest) => {
+            return request.operation === 'create' || request.operation === 'update';
+        });
+
+        requests.forEach((request: BatchReadWriteRequest) => {
+            if (request.operation === 'create') {
+                idToVersionId[`${request.resourceType}_${request.id}`] = '1';
+            }
+        });
+
+        await Promise.all(
+            createOrUpdates.map(async (request: BatchReadWriteRequest) => {
+                const { resource } = request;
+                const versionedLinksArray = this.versionedLinks[request.resourceType];
+                if (!versionedLinksArray) {
+                    return;
+                }
+                const flattenedResources: Record<string, string> = flatten(resource);
+                const references = Object.keys(flattenedResources).filter((key: string) => {
+                    return key.endsWith('.reference');
+                });
+                if (references.length === 0) {
+                    return;
+                }
+                const versionedLinksSet = new Set(versionedLinksArray);
+                await Promise.all(
+                    references.map(async (referencePath: string) => {
+                        if (!versionedLinksSet.has(referencePath)) {
+                            return;
+                        }
+                        let entryReference = flattenedResources[referencePath];
+                        const fullUrlMatch = entryReference.match(captureFullUrlParts);
+                        if (!fullUrlMatch) {
+                            return;
+                        }
+                        const resourceType = fullUrlMatch[2];
+                        const id = fullUrlMatch[3];
+                        let vid = fullUrlMatch[4];
+                        if (!vid) {
+                            const compoundId = `${resourceType}_${id}`;
+                            if (compoundId in idToVersionId) {
+                                vid = idToVersionId[compoundId];
+                            } else {
+                                const projectionExpression = 'meta';
+                                try {
+                                    const itemResponse: GenericResponse = await this.dynamoDbHelper.getMostRecentResource(
+                                        resourceType,
+                                        id,
+                                        projectionExpression,
+                                    );
+                                    const { meta } = itemResponse.resource;
+                                    vid = meta.versionId;
+                                } catch (e) {
+                                    logger.error('Failed to find most recent resource', e);
```

Comment:
what should be do in this case? swallow the error and move on? or fail the transaction?

Reply:
Good question. Nobody likes silent errors. It might hint at a programming error if those occur. It's probably safer to fail the tx.